src/puppetlabs/puppetdb/command.clj

```diff
@@ -257,12 +258,12 @@
   [{:keys [payload annotations]} {:keys [db]}]
   (let [certname (:certname payload)
         producer-timestamp (to-timestamp (:producer_timestamp payload (now)))
-        id  (:id annotations)
-        newer-record-exists? (fn [entity] (scf-storage/have-record-produced-after? entity certname producer-timestamp))]
+        id  (:id annotations)]
     (jdbc/with-transacted-connection db
       (when-not (scf-storage/certname-exists? certname)
         (scf-storage/add-certname! certname))
-      (if (not-any? newer-record-exists? [:catalogs :factsets :reports])
+      ;; Should this guard be moved to deactivate-node!?
```

Comment:
yes, it should. Would rather not leave these comments in the code though; instead, they can go into a tech debt ticket.